unittests/test_pipeline.py

```diff
@@ -457,6 +457,37 @@ def postfoo(self):
         assert '_RFM_PRE_SETUP' in os.environ
         assert '_RFM_POST_SETUP' in os.environ
 
+    def test_setup_hooks_in_compile_only_test(self):
+        @fixtures.custom_prefix('unittests/resources/checks')
+        class MyTest(rfm.CompileOnlyRegressionTest):
+            def __init__(self):
+                self.name = 'hellocheck_compile'
+                self.descr = 'C Hello World test'
+
+                # All available systems are supported
```

Comment:
No need for this comment, too. And remove it the blank line before as well. Let's keep this inline test as short and simple as possible.